agent/consul/rpc.go

```diff
@@ -370,16 +363,18 @@ func (s *Server) blockingQuery(queryOpts *structs.QueryOptions, queryMeta *struc
 		goto RUN_QUERY
 	}
 
-	// Restrict the max query time, and ensure there is always one.
-	if queryOpts.MaxQueryTime > maxQueryTime {
-		queryOpts.MaxQueryTime = maxQueryTime
-	} else if queryOpts.MaxQueryTime <= 0 {
-		queryOpts.MaxQueryTime = defaultQueryTime
+	if queryOpts.MaxQueryTime <= 0 {
+		queryOpts.MaxQueryTime = s.config.DefaultQueryTime
 	}
 
 	// Apply a small amount of jitter to the request.
 	queryOpts.MaxQueryTime += lib.RandomStagger(queryOpts.MaxQueryTime / jitterFraction)
```

Comment:
If MaxQueryTime is set to a very big value, jitter might be quite high.
Limited for now to 37s, but if the value is very high (for instance 24h), then value might be quite high... do you think we might put a max on it ? (such as 60s)?

Reply:
@pierresouchay: do you mind if I add the jitter feature in another pull request? The config is changing a lot and rebasing causes conflicts that are tedious to resolve. It would help if I kept the PRs small.